src/main/java/fr/xephi/authme/process/quit/AsynchronousQuit.java

```diff
@@ -82,8 +82,11 @@ public void processQuit(Player player) {
         playerCache.removePlayer(name);
 
         //always update the database when the player quit the game (if sessions are disabled)
-        if(!wasLoggedIn || !service.getProperty(PluginSettings.SESSIONS_ENABLED)) {
+        if(wasLoggedIn) {
             database.setUnlogged(name);
+            if(!service.getProperty(PluginSettings.SESSIONS_ENABLED)) {
```

Comment:
here and above, would be nice to have a space before reserved keywords like `if` `while` `for` etc.
Same for the `AsynchronousLogin` addition